src/schedule/ll_schedule.c

```diff
@@ -149,9 +150,26 @@ static void schedule_ll_clients_enable(struct ll_schedule_data *sch)
 
 static void schedule_ll_clients_reschedule(struct ll_schedule_data *sch)
 {
+	struct list_item *wlist;
+	struct list_item *tlist;
+	struct task *task;
+	uint64_t next_tick = UINT64_MAX;
+
+	/* update the last_tick */
+	sch->domain->last_tick = sch->domain->next_tick;
+
 	/* rearm only if there is work to do */
 	if (atomic_read(&sch->domain->total_num_tasks)) {
-		domain_set(sch->domain, sch->domain->last_tick);
+		/* traverse to set timer according to the earliest task */
```

Comment:
this should be atomic - i assume lock are held elsewhere and IRQs off ?

Reply:
> this should be atomic - i assume lock are held elsewhere and IRQs off ?

yes, sch->domain->lock is already hold in the caller schedule_ll_tasks_run().